packages/web/src/features/translation/TranslationView.tsx

```diff
@@ -112,6 +116,59 @@ export default function TranslationView() {
 
   const userCan = useAccessControl();
 
+  const handleKeyPress = useCallback(
+    (event: {
+      key: string;
+      shiftKey: boolean;
+      ctrlKey: boolean;
+      preventDefault: VoidFunction;
+      stopPropagation: VoidFunction;
+    }) => {
+      const watchKeys = ['ArrowUp', 'ArrowDown', 'Home', 'End'];
+      if (!event.ctrlKey || !watchKeys.includes(event.key)) {
+        return;
+      }
+      event.preventDefault();
+      event.stopPropagation();
+      switch (event.key) {
+        case 'ArrowUp':
+          navigate(
+            `/languages/${language}/verses/${decrementVerseId(verseId)}`
+          );
+          break;
+        case 'ArrowDown':
+          navigate(
+            `/languages/${language}/verses/${incrementVerseId(verseId)}`
+          );
+          break;
+        case 'Home':
+          if (event.shiftKey) {
+            console.log('FIRST VERSE!');
+          } else {
+            console.log('FIRST WORD!');
+          }
+          break;
+        case 'End':
+          if (event.shiftKey) {
+            console.log('LAST VERSE!');
+          } else {
+            console.log('LAST WORD!');
+          }
```

Comment:
I've got previous/next verse functionality working, I still need to implement the other 4 shortcuts. The first/last verse of book functionality should be easy, I just need to write the helper function that calculated the verse ID.

@arrocke Any hints for how to implement the first/last word shortcuts? How would I manually move focus to the correct `TranslateWord` components?

Reply:
This morning I learned how to use `forwardRef`! :smile: